freezeyt/middleware.py

```diff
@@ -1,14 +1,44 @@
 from typing import Iterable
 
-from freezeyt.compat import StartResponse, WSGIEnvironment, WSGIApplication
+from werkzeug.wrappers import Response
+from werkzeug.exceptions import NotFound, Forbidden
+from werkzeug.routing import Map, Rule
+from werkzeug.routing.exceptions import RequestRedirect
+from werkzeug.security import safe_join
+from werkzeug.utils import send_file
 
+from freezeyt.compat import StartResponse, WSGIEnvironment, WSGIApplication
 from freezeyt.mimetype_check import MimetypeChecker
+from freezeyt.extra_files import get_extra_files
 
 
 class Middleware:
     def __init__(self, app: WSGIApplication, config):
         self.app = app
         self.mimetype_checker = MimetypeChecker(config)
+        self.url_map = Map()
+        for url_part, kind, content_or_path in get_extra_files(config):
+            if '<' in url_part:
+                raise NotImplementedError("the extra file URL cannot include '<'")
+            if kind == 'content':
+                self.url_map.add(Rule(
+                    f'/{url_part}',
+                    endpoint='content',
+                    defaults={'content': content_or_path},
+                ))
+            elif kind == 'path':
+                self.url_map.add(Rule(
+                    f'/{url_part}',
+                    endpoint='path',
+                    defaults={'path': content_or_path, 'subpath': None},
+                ))
+                self.url_map.add(Rule(
+                    f'/{url_part}/<path:subpath>',
+                    endpoint='path',
+                    defaults={'path': content_or_path},
+                ))
```

Comment:
If I am right, we always get path to a file... rules are added in format from `get_extra_files` results. So it should be ensured all rules will have format of path with slashes.

I suppose we have configuration in format `url_part` url in web app which lead to static file and `path` physical location of file on server. Paths which are yielded from `get_url_parts_from_directory` should be already base + subpath why we are splitting again?

I have to leave now, I'll try to edit for better understandig later...
I give an example what I mean at least..
```
from werkzeug.routing import Map, Rule
import werkzeug.exceptions

# 'extra_files': {'static': {'copy_from': str(APP_DIR / 'static_dir')}}
# to extra_files_config goes goes {'static': {'copy_from': str(APP_DIR / 'static_dir')}}
# static as url_part and rest as content
url_map = Map()
results_from_get_extra_files = (
    ('a', 'static/a/a/a'),
    ('b/c/d', 'path/to/static/d')
)

for url_part, path in results_from_get_extra_files:
    url_map.add(
        Rule(
            f'/{url_part}',
            endpoint='path',
            defaults={'path': path},
        )
    )

print(url_map)
adapter = url_map.bind('foo.test')
for url in ('/a', '/b/c/d', '/a/'):
    try:
        print(url, adapter.match(url))
    except werkzeug.exceptions.NotFound:
        print(url, '(404)')
```